react/w-books/src/app/components/SignUp.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect, useState } from 'react';
+import { useForm } from 'react-hook-form';
+
+import logo from '../assets/logo_full_color.svg';
+
+export const SignUp = () => {
+  const [state, setState] = useState({
+    user: {
+      email: '',
+      password: '',
+      passwordConfirmation: '',
+      firstName: '',
+      lastName: '',
+      locale: 'en'
+    }
```

Comment:
I think that this would be cleaner without the `user`. Just leave the elements you need:
```
{
      email: '',
      password: '',
      passwordConfirmation: '',
      firstName: '',
      lastName: '',
      locale: 'en'
    }
```